qt_ui/windows/basemenu/QBaseMenu2.py

```diff
@@ -241,9 +241,47 @@ def closeEvent(self, close_event: QCloseEvent):
         GameUpdateSignal.get_instance().updateGame(self.game_model.game)
 
     def get_base_image(self):
+        try:
+            game = self.game_model.game
+        except:
+            game = None
+
         if self.cp.cptype == ControlPointType.AIRCRAFT_CARRIER_GROUP:
+            for ground_object in self.cp.ground_objects:
+                for group in ground_object.groups:
+                    for unit in group.units:
+                        if unit.type == "Forrestal":
```

Comment:
This should be factored out in `ControlPoint`. `ControlPoint.get_carrier_group_type`. Also factor out the common portion of that and `ControlPoint.get_carrier_group_name` into `ControlPoint.get_carrier_group`.

Reply:
Done. In Revision: 4de3fee2a32c2be3819ee8eb894f72d7c4795d37

Reply:
This also ended up fixing a bug in flightgroupspawner.py where mission.find_group(group_name) would fail, causing the mission generation to also not succeed.
![image](https://user-images.githubusercontent.com/89945461/145716705-50085c3b-5f72-49c3-b9b4-1df78a4ba6e5.png)